src/search.js

```diff
@@ -0,0 +1,231 @@
+"use strict";
+
+var path = require("path");
+var fs = require("fs");
+var moment = require("moment");
+var Promise = require("bluebird");
+var Sqlite = require("sqlite3");
+var fsextra = require("fs-extra");
+var es = require("event-stream");
+var Helper = require("./helper");
+
+try {
+	fsextra.ensureDirSync(Helper.getLogsPath());
+} catch (e) {
+	log.error("Unable to create log directory", e);
+	return;
+}
+
+var format = Helper.config.logs.format || "YYYY-MM-DD HH:mm:ss";
+
+var schema = [
+	"CREATE TABLE IF NOT EXISTS logs (user TEXT, network TEXT, chan TEXT, time INTEGER, nick TEXT, text TEXT)",
+	"CREATE VIRTUAL TABLE IF NOT EXISTS texts USING fts4(content=\"logs\", nick, text)",
```

Comment:
Any reason you need a virtual table? I think just `logs` should be fine.

Reply:
This is not a "normal" virtual table, but more the way to use full text search capabilities for ex. you can use "*" in the text and nick input fields.

See https://www.sqlite.org/fts3.html for more infos.